hcl2template/types.packer_config.go

```diff
@@ -16,150 +16,56 @@ type PackerConfig struct {
-	// 	configs, err := r.parsePostProcessor(i, v)
-	// 	if err != nil {
-	// 		errs = multierror.Append(errs, err)
-	// 		continue
-	// 	}
-
-	// 	// Parse the PostProcessors out of the configs
-	// 	pps := make([]*PostProcessor, 0, len(configs))
-	// 	for j, c := range configs {
-	// 		var pp PostProcessor
-	// 		if err := r.decoder(&pp, nil).Decode(c); err != nil {
-	// 			errs = multierror.Append(errs, fmt.Errorf(
-	// 				"post-processor %d.%d: %s", i+1, j+1, err))
-	// 			continue
-	// 		}
-
-	// 		// Type is required
-	// 		if pp.Type == "" {
-	// 			errs = multierror.Append(errs, fmt.Errorf(
-	// 				"post-processor %d.%d: type is required", i+1, j+1))
-	// 			continue
-	// 		}
-
-	// 		// Set the raw configuration and delete any special keys
-	// 		pp.Config = c
-
-	// 		// The name defaults to the type if it isn't set
-	// 		if pp.Name == "" {
-	// 			pp.Name = pp.Type
-	// 		}
-
-	// 		delete(pp.Config, "except")
-	// 		delete(pp.Config, "only")
-	// 		delete(pp.Config, "keep_input_artifact")
-	// 		delete(pp.Config, "type")
-	// 		delete(pp.Config, "name")
-
-	// 		if len(pp.Config) == 0 {
-	// 			pp.Config = nil
-	// 		}
-
-	// 		pps = append(pps, &pp)
-	// 	}
-
-	// 	result.PostProcessors = append(result.PostProcessors, pps)
-	// }
-
-	// // Gather all the provisioners
-	// if len(r.Provisioners) > 0 {
-	// 	result.Provisioners = make([]*Provisioner, 0, len(r.Provisioners))
-	// }
-	// for i, v := range r.Provisioners {
-	// 	var p Provisioner
-	// 	if err := r.decoder(&p, nil).Decode(v); err != nil {
-	// 		errs = multierror.Append(errs, fmt.Errorf(
-	// 			"provisioner %d: %s", i+1, err))
-	// 		continue
-	// 	}
-
-	// 	// Type is required before any richer validation
-	// 	if p.Type == "" {
-	// 		errs = multierror.Append(errs, fmt.Errorf(
-	// 			"provisioner %d: missing 'type'", i+1))
-	// 		continue
-	// 	}
-
-	// 	// Set the raw configuration and delete any special keys
-	// 	p.Config = v.(map[string]interface{})
-
-	// 	delete(p.Config, "except")
-	// 	delete(p.Config, "only")
-	// 	delete(p.Config, "override")
-	// 	delete(p.Config, "pause_before")
-	// 	delete(p.Config, "type")
-	// 	delete(p.Config, "timeout")
-
-	// 	if len(p.Config) == 0 {
-	// 		p.Config = nil
-	// 	}
-
-	// 	result.Provisioners = append(result.Provisioners, &p)
-	// }
-
-	// // If we have errors, return those with a nil result
-	// if errs != nil {
-	// 	return nil, errs
-	// }
+func (p *PackerConfig) GetBuilds() ([]packer.Build, hcl.Diagnostics) {
+	res := []packer.Build{}
+	var diags hcl.Diagnostics
 
-	return &result, nil
+	for _, build := range p.Builds {
+		for _, from := range build.Froms {
+			src := p.Sources[from]
+			builder, moreDiags := src.startBuilder()
+			diags = append(diags, moreDiags...)
+
+			pcb := &packer.CoreBuild{
+				Type:           src.Type,
+				Builder:        builder,
+				BuilderConfig:  nil, // TODO(azr): do we really need that ?
+				Hooks:          nil, // TODO(azr): what are hooks and do we really need those ?
```

Comment:
@SwampDragons; can a user set Hooks from a buildfile ? if not then I will probably unexport that field.

Reply:
nope, hooks are very much an internal-only thing.

Reply:
Ah, making this field unexported then !!